### DISCUSSION TITLE
Duplicate XAUTH logins when using multiple rightsubnet.

### DISCUSSION THREAD

**Discussion by jamison123** [strongswan/strongswan#891] [General]
I have the following config that successfully connects to a fortigate VPN which sends back a new PIN code SMS each time you log in, all routes work etc.

The server recently add a new subnet, 10.0.0.0/8, so I added the MY_VPN_ADDITIONAL part - and that too works just fine, except for one annoying thing.

The server sends me a new SMS login code twice, not just once. When I study the strongswan logs it's running the XAUTH twice, once for MY_VPN and again for MY_VPN_ADDITIONAL.

Is there any way to tell MY_VPN_ADDITIONAL to use the existing authentication that has already worked?

Many many thanks brains trust!

```


# Introduction to IPsec: http://www.ipsec-howto.org/x202.html
config setup
  charondebug = "dmn 1, mgr 1, ike 1, chd 1, job 1, cfg 1, knl 1, net 1, lib 1, esp 1"
  # charondebug = "dmn 1, mgr 1, ike 2, chd 1, job 1, cfg 3, knl 2, net 2, lib 1, esp 1"
  # charondebug = "dmn 4, mgr 4, ike 4, chd 4, job 4, cfg 4, knl 4, net 4, lib 4, esp 4"

conn MY_VPN
  type=tunnel
  keyexchange = ikev1
  ike = aes256-sha256-modp1536
  esp = aes256-sha256

  lifetime = 86400

  # You'll have to find out whether your FortiGate uses aggressive mode for
  # authentication. If it does, you must set "aggressive = yes" here to
  # connect successfully
  aggressive = yes

  right = 195.27.NNN.XX
  rightsubnet = 172.0.0.0/8
  rightid = 195.27.NNN.XX
  rightauth = psk
  # rightauth2 = xauth

  leftsourceip=%config
  leftauth = psk
  leftauth2 = xauth
  leftfirewall=yes
  lefthostaccess=yes
  xauth_identity = "YOUR_USERNAME"

  auto = start

conn MY_VPN_ADDITIONAL
  also=MY_VPN
  rightsubnet=10.0.0.0/8
```

**Comment by tobiasbrunner** [comment]
Please post the complete log (see #196 for recommended log settings).
